pkg/wasm/imagefetcher.go

```diff
@@ -22,20 +22,108 @@ import (
 	"crypto/tls"
 	"fmt"
 	"io"
+	"io/ioutil"
 	"path/filepath"
 	"reflect"
 	"strings"
+	"time"
+
+	"istio.io/istio/pkg/bootstrap/platform"
 
 	"github.com/docker/cli/cli/config/configfile"
 	dtypes "github.com/docker/cli/cli/config/types"
 	"github.com/google/go-containerregistry/pkg/authn"
 	"github.com/google/go-containerregistry/pkg/name"
 	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/google"
```

Comment:
please not couple vendor code with core path

Reply:
Suggest each platform provide a setup method in their own dir

Reply:
@hzxuzhonghu Please take a look into the revised codes.